metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py

```diff
@@ -707,6 +850,182 @@ def filter_nodes(self, all_nodes: List[DBTNode]) -> List[DBTNode]:
+            added_to_schema_resolver = False
+            if target_node_urn and schema_fields:
+                schema_resolver.add_raw_schema_info(
+                    target_node_urn, self._to_schema_info(schema_fields)
+                )
+                added_to_schema_resolver = True
+
+            # Run sql parser to infer the schema + generate column lineage.
+            sql_result = None
+            if node.compiled_code:
+                try:
+                    # Add CTE stops based on the upstreams list.
+                    preprocessed_sql = detach_ctes(
+                        node.compiled_code,
+                        platform=schema_resolver.platform,
+                        cte_mapping={
+                            cte_name: upstream_node.get_fake_ephemeral_table_name()
+                            for upstream_node in [
+                                all_nodes_map[upstream_node_name]
+                                for upstream_node_name in node.upstream_nodes
+                                if upstream_node_name in all_nodes_map
+                            ]
+                            if upstream_node.is_ephemeral_model()
+                            for cte_name in _get_dbt_cte_names(
+                                upstream_node.name, schema_resolver.platform
+                            )
+                        },
+                    )
+                except Exception as e:
+                    sql_result = SqlParsingResult.make_from_error(e)
+                else:
+                    sql_result = sqlglot_lineage(
+                        preprocessed_sql, schema_resolver=schema_resolver
+                    )
+
+            # Save the column lineage.
+            if self.config.include_column_lineage and sql_result:
+                # We only save the debug info here. We're report errors based on it later, after
+                # applying the configured node filters.
+                node.cll_debug_info = sql_result.debug_info
+
+                if sql_result.column_lineage:
+                    node.upstream_cll = [
+                        DBTColumnLineageInfo(
+                            upstream_dbt_name=target_platform_urn_to_dbt_name[
+                                upstream_column.table
+                            ],
+                            upstream_col=upstream_column.column,
+                            downstream_col=column_lineage_info.downstream.column,
+                        )
+                        for column_lineage_info in sql_result.column_lineage
+                        for upstream_column in column_lineage_info.upstreams
+                        # Only include the CLL if the table in in the upstream list.
+                        if target_platform_urn_to_dbt_name.get(upstream_column.table)
+                        and target_platform_urn_to_dbt_name[upstream_column.table]
+                        in node.upstream_nodes
```

Comment:
Can we not just do `target_platform_urn_to_dbt_name.get(upstream_column.table) in node.upstream_nodes`? Can `upstream_nodes` contain `None`?

Reply:
yup that works